### ISSUE TITLE
nullpointer at com.dtolabs.rundeck.core.common.NodeEntryImpl.setTags after update from 2.7.3 to 2.9.3

### ISSUE THREAD

**Issue by hoodoomeister** [rundeck/rundeck#2730]
Issue type: Bug report/Enhancement Request

**My Rundeck detail**

* Rundeck version: 2.9.3
* install type: (rpm,deb,war,launcher?) launcher
* OS Name/version: mac
* DB Type/version: mysql? postgres? h2?..

**Expected Behavior**
Should all nodes and jobs be uploaded and rundeck run without exceptions.

**Actual Behavior**
InvocationTargetException and NullPointerException are returned after jobs and nodes upload

java.lang.reflect.InvocationTargetException
 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.lang.reflect.Method.invoke(Method.java:498)
 	at org.apache.commons.beanutils.PropertyUtilsBean.invokeMethod(PropertyUtilsBean.java:2170)
 	at org.apache.commons.beanutils.PropertyUtilsBean.setSimpleProperty(PropertyUtilsBean.java:2151)
 	at org.apache.commons.beanutils.PropertyUtilsBean.setNestedProperty(PropertyUtilsBean.java:1957)
 	at org.apache.commons.beanutils.PropertyUtilsBean.setProperty(PropertyUtilsBean.java:2064)
 	at org.apache.commons.beanutils.BeanUtilsBean.setProperty(BeanUtilsBean.java:1017)
 	at org.apache.commons.beanutils.BeanUtilsBean.populate(BeanUtilsBean.java:830)
 	at org.apache.commons.beanutils.BeanUtils.populate(BeanUtils.java:433)
 	at com.dtolabs.rundeck.core.common.NodeEntryFactory.createFromMap(NodeEntryFactory.java:95)
 	at com.dtolabs.rundeck.core.common.NodesYamlParser.parse(NodesYamlParser.java:100)
 	at com.dtolabs.rundeck.core.resources.format.ResourceYamlFormatParser.parseDocument(ResourceYamlFormatParser.java:65)
 	at com.dtolabs.rundeck.core.resources.FileResourceModelSource.loadNodes(FileResourceModelSource.java:335)
 	at com.dtolabs.rundeck.core.resources.FileResourceModelSource.getNodes(FileResourceModelSource.java:275)
 	at com.dtolabs.rundeck.core.resources.FileResourceModelSource.getNodes(FileResourceModelSource.java:258)
 	at com.dtolabs.rundeck.core.resources.FileResourceModelSource.parseFile(FileResourceModelSource.java:432)
 	at com.dtolabs.rundeck.core.resources.ScriptResourceUtil.executeScript(ScriptResourceUtil.java:152)
 	at com.dtolabs.rundeck.core.resources.ScriptResourceModelSource.getNodes(ScriptResourceModelSource.java:156)
 	at com.dtolabs.rundeck.core.resources.ExceptionCatchingResourceModelSource.getNodes(ExceptionCatchingResourceModelSource.java:57)
 	at com.dtolabs.rundeck.core.common.ProjectNodeSupport.getNodeSet(ProjectNodeSupport.java:113)
 	at com.dtolabs.rundeck.core.common.ProjectNodeSupport$ProjectNodesSource.getNodes(ProjectNodeSupport.java:327)
 	at com.dtolabs.rundeck.core.resources.ExceptionCatchingResourceModelSource.getNodes(ExceptionCatchingResourceModelSource.java:57)
 	at com.dtolabs.rundeck.core.resources.ResourceModelSource$getNodes.call(Unknown Source)
 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:117)
 	at rundeck.services.nodes.CachedProjectNodes.reloadNodeSet(CachedProjectNodes.groovy:42)
 	at rundeck.services.nodes.CachedProjectNodes$reloadNodeSet.call(Unknown Source)
 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:117)
 	at rundeck.services.NodeService$_loadNodes_closure3.doCall(NodeService.groovy:265)
 	at rundeck.services.NodeService$_loadNodes_closure3.doCall(NodeService.groovy)
 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.lang.reflect.Method.invoke(Method.java:498)
 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1218)
 	at groovy.lang.ExpandoMetaClass.invokeMethod(ExpandoMetaClass.java:1125)
 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1027)
 	at groovy.lang.Closure.call(Closure.java:414)
 	at groovy.lang.Closure.call(Closure.java:408)
 	at com.codahale.metrics.Timer.time(Timer.java:99)
 	at com.codahale.metrics.Timer$time$0.call(Unknown Source)
 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
 	at com.codahale.metrics.Timer$time$0.call(Unknown Source)
 	at MetricswebGrailsPlugin$_addDynamicMetricMethods_closure26.doCall(MetricswebGrailsPlugin.groovy:190)
 	at sun.reflect.GeneratedMethodAccessor354.invoke(Unknown Source)
 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.lang.reflect.Method.invoke(Method.java:498)
 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1218)
 	at groovy.lang.ExpandoMetaClass.invokeMethod(ExpandoMetaClass.java:1125)
 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1027)
 	at groovy.lang.Closure.call(Closure.java:414)
 	at org.codehaus.groovy.runtime.metaclass.ClosureStaticMetaMethod.invoke(ClosureStaticMetaMethod.java:62)
 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
 	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoMetaMethodSiteNoUnwrap.invoke(PogoMetaMethodSite.java:233)
 	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.call(PogoMetaMethodSite.java:71)
 	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:141)
 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callSafe(AbstractCallSite.java:101)
 	at rundeck.services.NodeService$_loadNodes_closure4.doCall(NodeService.groovy:272)
 	at rundeck.services.NodeService$_loadNodes_closure4.doCall(NodeService.groovy)
 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.lang.reflect.Method.invoke(Method.java:498)
 	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
 	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1218)
 	at groovy.lang.ExpandoMetaClass.invokeMethod(ExpandoMetaClass.java:1125)
 	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1027)
 	at groovy.lang.Closure.call(Closure.java:414)
 	at groovy.lang.Closure.call(Closure.java:408)
 	at groovy.lang.Closure.run(Closure.java:495)
 	at org.springframework.core.task.SimpleAsyncTaskExecutor$ConcurrencyThrottlingRunnable.run(SimpleAsyncTaskExecutor.java:251)
 	at java.lang.Thread.run(Thread.java:745)
 Caused by: java.lang.NullPointerException
 	at com.dtolabs.rundeck.core.common.NodeEntryImpl.setTags(NodeEntryImpl.java:127)

**How to reproduce Behavior**
1. Run rundeck from docker-compose.
2. Watch container logs.

**Comment by gschueler** [comment]
It seems like you have a Script resource model source expecting your script to produce YAML data.  What the output of your script?

**Comment by hoodoomeister** [comment]
I checked scripts as you suggested and you were right. For one python script it returns above stacktrace, but produced yaml is well formatted anyway. I checked it on some online yaml validator. 
This script generates following yaml:
---
some_node1:
  node: some_node1
  hostname: some_node1
  tags:
  - "key1:value1"
  - "key2:value2"
some_node2:
  node: some_node2
  hostname: some_node2
  tags:
  - "key1:value12"
  - "key2:value22"

...etc
